### PR TITLE
Improve documentation

### LINKED ISSUE #39: Document configuration options
Including how to configure API access, as per #35

### LINKED ISSUE #40: Current example without leader election
It seems that the previous `examples/calico-policy-agent.yaml` have a leader election container, but now it's missing? We are doing k8s HA master and want to deploy several controllers.

### PR DESCRIPTION
**Pull request by caseydavenport**
Fixes #40 
Fixes #39

### CODE REVIEW

File: examples/policy-controller.yaml [hunk 1]
```diff
@@ -1,16 +1,22 @@
 # Create this manifest using kubectl to deploy
 # the Calico policy controller on Kubernetes.
+# It deploys a single instance of the policy controller.
 apiVersion: v1
```
Comment (caseydavenport) [inline examples/policy-controller.yaml]: This needs to be extensions/v1beta1

File: configuration.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Policy Controller Configuration
```
Comment (djosborne) [inline configuration.md]: Calico Policy Controller Configuration?

At first I was wondering why we were documenting k8s policy controller, but then realized it is ours

File: configuration.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Policy Controller Configuration 
+
+The policy controller is primarily configured through environment variables.  When running
+the policy controller as a Kubernetes pod, this is accomplished through the pod manifest `env`
+section.
+
+## The kube-policy-controller container
+
+### Etcd configuration
```
Comment (djosborne) [inline configuration.md]: Etcd Settings

We're not configuring etcd here, we're configuring calico with etcd's settings.

Comment (djosborne) [inline configuration.md]: Maybe something like `Configure etcd Access`, `Configure Kubernetes API Access`??

File: configuration.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+The [leader election container](./election/Dockerfile) is an optional sidecar container which performs leader election using the Kubernetes API.
+This ensures that only a single instance of the policy controller is ever active.  
+
+The leader election container is only recommended when running the policy controller as a static pod in a multi-master deployment. 
+
+However, it is instead recommended to use a `ReplicaSet` with a single replica to ensure that one instance
+will always be running without need for leader election.
+
+### Kubernetes API access
+The leader election container also needs Kubernetes API access, which can be configured through a `kubeconfig` file placed in
```
Comment (djosborne) [inline configuration.md]: Placed in **it's** root directory ?

Comment (caseydavenport) [inline configuration.md]: Probably "the root directory of the container"

**Comment by djosborne** [comment]
Few things, then LGTM

**Pull request merged**